src/oncall/scheduler/default.py

```diff
@@ -123,50 +123,52 @@ def get_busy_user_by_event_range(self, user_ids, team_id, events, cursor):
             query_params += [sub['subscription_id'], sub['role_id']]
 
         query = '''
-                SELECT DISTINCT `user_id` FROM `event`
+                SELECT DISTINCT `user_id` FROM `%s`
                 WHERE `user_id` in %%s AND (%s) AND (%s)
-                ''' % (' OR '.join(range_check), ' OR '.join(team_check))
+                ''' % (table_name, ' OR '.join(range_check), ' OR '.join(team_check))
 
         cursor.execute(query, query_params)
         return [r['user_id'] for r in cursor.fetchall()]
 
-    def find_least_active_user_id_by_team(self, user_ids, team_id, start_time, role_id, cursor):
+    def find_least_active_user_id_by_team(self, user_ids, team_id, start_time, role_id, cursor, table_name):
         '''
         Of the people who have been oncall before, finds those who haven't been oncall for the longest. Start
         time refers to the start time of the event being created, so we don't accidentally look at future
         events when determining who was oncall in the past. Done on a per-role basis, so we don't take manager
         or vacation shifts into account
         '''
-        cursor.execute('''
-            SELECT `user_id`, MAX(`end`) AS `last_end` FROM `event`
-            WHERE `team_id` = %s AND `user_id` IN %s AND `end` <= %s
+        user_id_list = ', '.join(map(str, user_ids))
+        query = '''
+            SELECT `user_id`, MAX(`end`) AS `last_end` FROM `%s`
+            WHERE `team_id` = %s AND `user_id` IN (%s) AND `end` <= %s
             AND `role_id` = %s
             GROUP BY `user_id`
-            ''', (team_id, user_ids, start_time, role_id))
+            ''' % (table_name, team_id, user_id_list, start_time, role_id)
```

Comment:
Can these be interpolated using cursor.execute?

Reply:
`table_name` can't be, but the rest should be to avoid SQL injection